src/main.cpp

```diff
@@ -2176,6 +2176,14 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_CHECKLOCKTIMEVERIFY;
     }
 
+    // Start enforcing BIP68 (sequence locks) and CHECKSEQUENCEVERIFY (BIP112) for block.nVersion=5
+    // blocks, when 75% of the network has upgraded:
+    int nLockTimeFlags = 0;
+    if (block.nVersion >= 5 && IsSuperMajority(5, pindex->pprev, chainparams.GetConsensus().nMajorityEnforceBlockUpgrade, chainparams.GetConsensus())) {
```

Comment:
This should be something like

    if ((block.nVersion & 15) >= 5 && IsSuperMajority... )

In order to avoid misinterpreting Version Bits blocks as votes for BIP112. IsSuperMajority would need to be modified slightly as well.

Once this has been done, Bitcoin Classic, XT, and Unlimited can use block versions like 0x300000005 to be able to vote for BIP109 and BIP112 simultaneously.